src/test/functional_tests.rs

```diff
@@ -11,6 +11,44 @@ use lightning::util::persist::KVStore;
 
 use std::sync::Arc;
 
+#[test]
+fn persist_peer_from_inbound_channel() {
+	// Channel A open a channel to B,
+	// B should persist A as a peer by adding there
+	// node_id to peer_store
+	let (bitcoind, electrsd) = setup_bitcoind_and_electrsd();
+	println!("== Node A ==");
+	let esplora_url = format!("http://{}", electrsd.esplora_url.as_ref().unwrap());
+	let config_a = random_config();
+	let mut builder_a = NodeBuilder::from_config(config_a);
+	builder_a.set_esplora_server(esplora_url.clone());
+	let node_a = builder_a.build().unwrap();
+	node_a.start().unwrap();
+
+	println!("\n== Node B ==");
+	let mut config_b = random_config();
+	config_b.trusted_peers_0conf = vec![node_a.node_id()];
+	let mut builder_b = NodeBuilder::from_config(config_b.clone());
+	builder_b.set_esplora_server(esplora_url.clone());
+	let node_b = builder_b.build().unwrap();
+	node_b.start().unwrap();
+	open_channel(&node_a, &node_b, &bitcoind, &electrsd, true);
+	let node_b_id = node_b.node_id();
+	let node_a_id = node_a.node_id();
+	node_b.stop().unwrap();
+	assert_eq!(node_b.stop(), Err(Error::NotRunning));
+	node_b.start().unwrap();
+	node_b.sync_wallets().unwrap();
+	assert_eq!(node_b.is_running(), true);
+	assert_eq!(node_b.node_id(), node_b_id);
+	assert_eq!(node_a.is_running(), true);
+	let node_b_peers = node_b.list_peers();
+	let node_b_peers = node_b_peers.get(0).unwrap();
+	assert_eq!(node_b_peers.node_id, node_a_id);
+	assert_eq!(node_b_peers.is_persisted, true);
+	assert_eq!(node_b_peers.is_connected, true);
```

Comment:
this assertion is failing, I tried to debug it and when I restart the node, it tries to reconnect to Peer A but returns ConnectionRefused, specifically it fails here https://github.com/lightningdevkit/ldk-node/pull/170/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R584

Reply:
any hints why this can happen?